### ISSUE TITLE
Team repo workflow best practices?

### ISSUE THREAD

**Issue by tlylt** [nus-cs2103-AY2021S2/forum#174]
Hi guys, I am trying to update the developer guide of our team repo and I am faced with an issue.

Suppose now that I have done the task of `remove all irrelevant content of the developer guide inherited from AB3`, and I created a PR from my `update-developer-guide` branch. Now while waiting for my teammates to approve it, I want to start working on adding user stories to the developer guide. Should I 

1.work on a new commit that is under the existing PR branch (`update-developer-guide`)? 
Or
2.create a new branch (`update-developer-guide-user-stories`) and make my changes there?

My concern with No.1 is that if I have completed my second task of updating user stories before the PR gets merged, the PR will then complete two tasks (remove irrelevant content and add user stories). Not sure if this is fine and I will also need to update the PR description. And if the PR gets merged before I complete my second task, I will make another PR will the existing branch, not sure if this is a good practice.

My concern with No.2 is that I might potentially have to make multiple branches trying to do similar things, not sure if this is a good practice.

Any thoughts on the above or if you would like to share your workflow? Thanks in advanced!

**Comment by damithc** [comment]
In general:
* If the two changesets are unlikely to cause a lot of conflicts, you can do them in parallel. It doesn't matter which PR is merged first.
* If you anticipate lot of conflicts between the two changesets, or one depends on the other, it makes more sense to wait until one is merged before starting on the other. 

In this particular case, get the first PR merged quickly as it is one of those initial morphing tasks that are better done early and quickly because most subsequent work will depend on it. Ask others to review quickly (no need for everyone to review though, just 1-2 reviews will do) so that it can be merged quickly.

**Comment by tlylt** [comment]
> In general:
> 
> * If the two changesets are unlikely to cause a lot of conflicts, you can do them in parallel. It doesn't matter which PR is merged first.
> * If you anticipate lot of conflicts between the two changesets, or one depends on the other, it makes more sense to wait until one is merged before starting on the other.
> 
> In this particular case, get the first PR merged quickly as it is one of those initial morphing tasks that are better done early and quickly because most subsequent work will depend on it. Ask others to review quickly (no need for everyone to review though, just 1-2 reviews will do) so that it can be merged quickly.

Got it, thanks prof!
